app_test.go

```diff
@@ -2265,3 +2265,239 @@ func TestWhenExitSubCommandWithCodeThenAppQuitUnexpectedly(t *testing.T) {
 		t.Errorf("exitCodeFromOsExiter valeu should be %v, but its value is %v", testCode, exitCodeFromExitErrHandler)
 	}
 }
+
+func TestDuplicateCommandName(t *testing.T) {
+	duplicateCommandName := "hello"
+	os.Args = []string{"say", duplicateCommandName}
+	app := NewApp()
+	app.Name = "say"
+	app.Commands = []Command{
+		{
+			Name: duplicateCommandName,
+			Action: func(c *Context) error {
+				fmt.Println("Hello")
+				return nil
+			},
+		},
+		{
+			Name: duplicateCommandName,
+			Action: func(c *Context) error {
+				fmt.Println("Hello duplicate")
+				return nil
+			},
+		},
+	}
+
+	err := app.Run(os.Args)
+	if err == nil {
+		t.Fatalf("expected to receive error from Run, got none")
+	}
+
+	if err.Error() != fmt.Sprintf("Your app contains multiple commands with the Name %q. Having multiple commands with the same name results in ambiguous behavior, so please make sure each command in your app has a unique name.", duplicateCommandName) {
```

Comment:
⭐⭐⭐ _(must change)_ Put error message text into a variable so that it's not copy pasted into the test